File: php/features.php
<?php
declare(strict_types=1);

// IinLookup SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IinLookupFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IinLookupBaseFeature();
            case "test":
                return new IinLookupTestFeature();
            default:
                return new IinLookupBaseFeature();
        }
    }
}

File: py/features.py
# IinLookup SDK feature factory

from feature.base_feature import IinLookupBaseFeature
from feature.test_feature import IinLookupTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IinLookupBaseFeature(),
        "test": lambda: IinLookupTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
